pkg/cluster/node.go

```diff
@@ -78,28 +85,36 @@ func (n *NodeInfo) populateNodeIpList() error {
 		return fmt.Errorf("can't fetch node list: %+v ", err)
 	}
 
+	if len(nodeList.Items) == 0 {
+		return fmt.Errorf("can't fetch node list: %+v ", err)
+	}
+
 	nodeIpArr := []corev1.NodeAddress{}
 	for i := 0; i < len(nodeList.Items); i++ {
 		nodeIpArr = append(nodeIpArr, nodeList.Items[i].Status.Addresses...)
 	}
 	for i := 0; i < len(nodeIpArr); i++ {
 		if nodeIpArr[i].Type == NodeExternalIP {
-			n.ExternalIP = append(n.ExternalIP, nodeIpArr[i].Address)
+			n.ExternalIPList = append(n.ExternalIPList, nodeIpArr[i].Address)
 		}
 	}
 	return err
 }
 
 func GetNodeIP(client client.Client) (string, error) {
-	nodeInfo := &NodeInfo{
-		Client: client,
-	}
-	nodeIPs, err := nodeInfo.getNodeExternalIpList(client)
-	if err != nil {
+	nodeInfo.Client = client
+	nodeIPs, err := nodeInfo.getNodeExternalIpList()
+	if err != nil || len(nodeIPs) == 0 {
 		log.Error(err, "Getting NodeIP From kube-api-server")
 		os.Exit(1)
```

Comment:
Why are we exiting the proc here? The caller of the func should decide if the proc should exit or not.

Reply:
sure, I will make changes